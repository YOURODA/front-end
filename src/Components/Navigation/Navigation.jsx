import React from "react";
import { List, ListItem, ListItemText } from "@material-ui/core";
import { withRouter, NavLink } from "react-router-dom";
import styled from "styled-components";

const ContainerDiv = styled.div`
    width: ${props => (props.width ? props.width : `100%`)};
    margin: 0 auto;
`;

ContainerDiv.displayName = "ContainerDiv";

const StyledLink = styled(NavLink)`
    text-decoration: none;
    position: relative;
    color: grey;
    transition: color 0.2s;
    &:hover {
        color: #fff;
    }
    &.active {
        color: #fff;
        &::before {
            content: "";
            height: 2px;
            background-color: #1db954;
            width: 30px;
            position: absolute;
            bottom: -6px;
            display: block;
            left: 50%;
            margin-left: -15px;
        }
    }
`;

StyledLink.displayName = "StyledLink";

const StyledList = styled(List)`
    display: flex;
    flex-direction: row;
    padding: 0;
    margin: 0 auto;
`;

StyledList.displayName = "StyledList";

const StyledListItem = styled(ListItem)`
    && {
        text-align: center;
    }
`;

StyledListItem.displayName = "StyledListItem";

export const Navigation = props => (
    <ContainerDiv width={props.width}>
        <StyledList>
            {props.items && props.items.length > 0
                ? props.items.map(item => (
                      <StyledListItem key={item.text}>
                          <ListItemText>
                              <StyledLink to={item.link}>
                                  {item.text}
                              </StyledLink>
                          </ListItemText>
                      </StyledListItem>
                  ))
                : null}
        </StyledList>
    </ContainerDiv>
);

export default withRouter(Navigation);
