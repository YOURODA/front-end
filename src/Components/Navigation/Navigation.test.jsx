import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import { Navigation } from "./Navigation";

configure({ adapter: new Adapter() });

describe("Navigation", () => {
    it("should render two navigation links in a list", () => {
        const wrapper = renderer
            .create(
                <BrowserRouter>
                    <Navigation
                        items={[
                            { text: "test1", link: "test1" },
                            { text: "test2", link: "test2" }
                        ]}
                    />
                </BrowserRouter>
            )
            .toJSON();
        expect(wrapper).toMatchSnapshot();
    });

    it("should render the text it was given as prop", () => {
        const wrapper = shallow(
            <Navigation items={[{ text: "test1", link: "test1" }]} />
        );
        expect(wrapper.find("StyledLink").props().children).toEqual("test1");
    });

    it("should link the user to the route it received from props", () => {
        const wrapper = shallow(
            <Navigation items={[{ text: "test", link: "/test" }]} />
        );
        expect(wrapper.find("StyledLink").props().to).toEqual("/test");
    });
});
