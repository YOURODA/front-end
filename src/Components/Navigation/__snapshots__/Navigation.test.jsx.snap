// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation should render two navigation links in a list 1`] = `
<div
  className="sc-bdVaJa ljTGkq"
>
  <ul
    className="MuiList-root-1 MuiList-padding-2 sc-htpNat gByEcQ"
  >
    <li
      className="MuiListItem-root-5 MuiListItem-default-8 MuiListItem-gutters-12 sc-bxivhb WRdkv"
      disabled={false}
    >
      <div
        className="MuiListItemText-root-16"
      >
        <span
          className="MuiTypography-root-22 MuiTypography-subheading-29 MuiListItemText-primary-19"
        >
          <a
            aria-current={null}
            className="sc-bwzfXH buioaA"
            href="/test1"
            onClick={[Function]}
          >
            test1
          </a>
        </span>
      </div>
    </li>
    <li
      className="MuiListItem-root-5 MuiListItem-default-8 MuiListItem-gutters-12 sc-bxivhb WRdkv"
      disabled={false}
    >
      <div
        className="MuiListItemText-root-16"
      >
        <span
          className="MuiTypography-root-22 MuiTypography-subheading-29 MuiListItemText-primary-19"
        >
          <a
            aria-current={null}
            className="sc-bwzfXH buioaA"
            href="/test2"
            onClick={[Function]}
          >
            test2
          </a>
        </span>
      </div>
    </li>
  </ul>
</div>
`;
